src/components/modals/ModalForm.tsx

```diff
@@ -137,8 +152,12 @@ export const ModalForm = forwardRef<HTMLDivElement, ModalFormProps>((props, ref)
   const defaultId = `${useId()}modal-form`;
   const formId = id ?? defaultId;
 
+  const ModalComponent = isDestructive ? ModalAlert : Modal;
+
   return (
-    <Modal
+    <ModalComponent
+      // description prop is mandatory for ModalAlert but not for Modal, force typing to the most discriminant type to avoid duplicating a lot of code
+      description={description as ModalAlertPropsSubset['description']}
```

Comment:
There is a way to make this typing work correctly by doing something like:

```ts
if (modalProps.isDestructive) {
  const { isDestructive, ...modalAlertProps} = modalProps;
  return <ModalAlert {...modalAlertProps} .../>
}

return <Modal {...modalProps} .../>
```

Doing this correctly switches the type of `modalProps` based on the value of `isDestructive`. But then it implies duplicating all the code that we pass to the Modal components... So I figured I prefer not have the duplication and force a type. But I'm open to suggestions if you have better ideas!

Reply:
Maybe a stupid question, but can't we use a ModalAlert all the time?
Semantically it makes less sense, but if it provides the structure we need :shrug:

Reply:
I don't think we should, because the modal alert also comes with a different role and interrupts the user.